### ISSUE TITLE
iOS 8 broken orientation with ofxiOSImagePicker

### ISSUE THREAD

**Issue by ofTheo** [openframeworks/openFrameworks#3474]
If you have hardware orientation enabled on iOS 8 and the app orientated in landscape the camera interface will show in landscape but the image view inside it will be 90 degrees off. So horizontally panning of the device will result in a vertical panning of the cameras view. 

To replicate in the master branch of examples/ios/ImagePickerExample change main.mm to: 

	ofAppiOSWindow * window = new ofAppiOSWindow();
    
    window->enableHardwareOrientation();
    window->enableOrientationAnimation();

	ofSetupOpenGL(window, 1024,768, OF_FULLSCREEN);			// <-------- setup the GL context

	ofRunApp(new ofApp);

and add:
	```ofSetOrientation(OF_ORIENTATION_90_LEFT);```

to ofApp::setup  

I suspect its something to do with the layer trying to handle orientation itself, but inheriting orientation from the main window as well.  

Also the switch camera interface buttons are 90 degrees off as well. 

edit: this is a partial fix - it changes the button interface to portrait. A better approach would be if the landscape interface could be made to work without the camera image being rotated 90 degrees. 

```
//--------------------------------------------------------------
- (BOOL) openCamera:(int)camera {
	if(cameraIsAvailable) {		
		_imagePicker.sourceType = UIImagePickerControllerSourceTypeCamera;
		if(camera == 0) {
			_imagePicker.cameraDevice = UIImagePickerControllerCameraDeviceRear;
        } else {
			if([UIImagePickerController isCameraDeviceAvailable:UIImagePickerControllerCameraDeviceFront]) {
				_imagePicker.cameraDevice = UIImagePickerControllerCameraDeviceFront;
            } else {
				_imagePicker.cameraDevice = UIImagePickerControllerCameraDeviceRear;
            }
		}

		[[[UIApplication sharedApplication] keyWindow] addSubview:_imagePicker.view];
		
		//On iOS 8 orientation is messed up if hardware orientation is enabled and the device is in landscape mode.
		//this orientates everything correctly but results in the UI being portrait.
		//this is not a perfect fix as it would be better to have the interface be in landscape.
		int order = [[[UIDevice currentDevice] systemVersion] compare:@"8.0" options:NSNumericSearch];
		if( order == NSOrderedSame || order == NSOrderedDescending ){
			UIInterfaceOrientation orientation = [UIApplication sharedApplication].statusBarOrientation;
			if(orientation == UIInterfaceOrientationLandscapeLeft) {
				_imagePicker.view.transform = CGAffineTransformMakeRotation(PI*0.5);
			} else if(orientation == UIInterfaceOrientationLandscapeRight) {
				_imagePicker.view.transform = CGAffineTransformMakeRotation(-PI*0.5);
			} else if(orientation == UIInterfaceOrientationPortraitUpsideDown) {
				_imagePicker.view.transform = CGAffineTransformMakeRotation(PI);
			}
			_imagePicker.view.frame = [UIApplication sharedApplication].delegate.window.frame; 
		}
	
		return true;
	} else {
		return false;
	}
}
```

**Comment by bilderbuchi** [comment]
@ofTheo is this maybe related to #3158?

**Comment by danoli3** [comment]
Tested this and it seems fixed now after #3865

**Comment by danoli3** [comment]
The real issue now is with the image content being displayed in the incorrect way.
Issue here: https://github.com/openframeworks/openFrameworks/blob/master/addons/ofxiOS/src/utils/ofxiOSImagePicker.mm#L334

I believe ```UIInterfaceOrientationLandscapeLeft``` and ```UIInterfaceOrientationLandscapeRight``` are inversed.

To be clear: 
- ```UIInterfaceOrientationLandscapeLeft``` = ```OF_ORIENTATION_90_RIGHT```
- ```UIInterfaceOrientationLandscapeRight``` = ```OF_ORIENTAITON_90_LEFT```

This is because the OF_ORIENTATION is based off the UIDeviceOrientation, not the UIInterfaceOrientation, in which this left / right landscape naming convention is reversed.

Definition:
```
// Note that UIInterfaceOrientationLandscapeLeft is equal to UIDeviceOrientationLandscapeRight (and vice versa).
// This is because rotating the device to the left requires rotating the content to the right.
typedef NS_ENUM(NSInteger, UIInterfaceOrientation) {
    UIInterfaceOrientationUnknown            = UIDeviceOrientationUnknown,
    UIInterfaceOrientationPortrait           = UIDeviceOrientationPortrait,
    UIInterfaceOrientationPortraitUpsideDown = UIDeviceOrientationPortraitUpsideDown,
    UIInterfaceOrientationLandscapeLeft      = UIDeviceOrientationLandscapeRight,
    UIInterfaceOrientationLandscapeRight     = UIDeviceOrientationLandscapeLeft
};
```
